### ISSUE TITLE
Wrong results with TikZ-positioning and  fill stretch image

### ISSUE THREAD

**Issue by nignasi** [T-F-S/tcolorbox#23]
It seems that when filling a node with `fill stretch image`, the use of positioning options breaks correct position of filling image.

    \documentclass{article} 
    \usepackage[most]{tcolorbox} 
    \usetikzlibrary{positioning}

    \begin{document}
    \begin{tikzpicture}
      \node[draw=red, minimum size=1cm, fill stretch image=goldshade.png] (a) {};
      \node[draw=red, minimum size=1cm, right=of a, fill stretch image=goldshade.png]  {};
      \node[draw=red, minimum size=1cm, above right=of a, fill stretch image=goldshade.png]  {};
      \node[draw=red, minimum size=1cm, above =of a, fill stretch image=goldshade.png]  {};
      \node[draw=red, minimum size=1cm, above left =of a, fill stretch image=goldshade.png]  {};
      \node[draw=red, minimum size=1cm, left =of a, fill stretch image=goldshade.png]  {};
      \node[draw=red, minimum size=1cm, below left=of a, fill stretch image=goldshade.png]  {};
      \node[draw=red, minimum size=1cm, below=of a, fill stretch image=goldshade.png]  {};
      \node[draw=red, minimum size=1cm, below right=of a, fill stretch image=goldshade.png]  {};
    \end{tikzpicture}
    \end{document}

![dibuix](https://cloud.githubusercontent.com/assets/2562430/24497584/3dced3d6-153c-11e7-8faf-399814e9136d.png)

**Comment by T-F-S** [comment]
The positioning options push through to the embedded node for the picture.

I think I can avoid the problem by using lower level `pgf` commands instead of a TikZ `node`. I expect that this problem also appears with some other of the filling commands.

The following patch for `fill stretch image` seems to work:

```latex
\tikzset{
  fill stretch image/.style={path picture={%
    \tcb@process@ppbb@wh%
    \pgftext[x=\the\dimexpr\pgf@xa+\tcb@ppbb@width/2\relax,y=\pgf@ya+\tcb@ppbb@height/2]
    {\expandafter\includegraphics\expandafter[\tcb@tikz@fill@graphics@options,%
      width=\the\dimexpr\tcb@tikz@fill@graphics@scale\dimexpr\tcb@ppbb@width\relax,%
      height=\the\dimexpr\tcb@tikz@fill@graphics@scale\dimexpr\tcb@ppbb@height\relax]{#1}}%
  }},
}
```

If this patch works out well, I intend to replace the embedded nodes by something like that.

**Comment by nignasi** [comment]
The patch works for me in my example.

**Comment by T-F-S** [comment]
Corrected with version `4.03`.
